compiler/noirc_evaluator/src/acir/acir_context/mod.rs

```diff
@@ -1029,6 +1029,30 @@ impl<F: AcirField, B: BlackBoxFunctionSolver<F>> AcirContext<F, B> {
         let zero = self.add_constant(F::zero());
         let one = self.add_constant(F::one());
 
+        // When dividing the twos complement lhs and rhs, we subtract one from the bit size
+        // to accurately represent the unsigned values we are dividing.
+        // We then internally constrain `q < 2^{max_q_bits}` during euclidean division.
+        // If we do not special case for binary types of one bit, we will range constrain the resulting
+        // quotient to be zero bits, which will cause inadvertent failures.
+        // We could theoretically still use this algorithm as is by incrementing the bit size by one,
+        // but the special case is very simple and then we don't mix the degenerate type
+        // into our general signed division algorithm.
+        //
+        // We know that a signed integer of bit size one can only be -1 or 0
+        // Thus, we know the only valid signed division can be 0 / -1.
+        // 0 / -1, valid as the quotient is 0
+        // -1 / -1, invalid as the quotient is 1, which cannot be represented in an i1
+        // All divisions by zero are invalid
+        if bit_size == 1 {
+            let lhs_is_zero = self.eq_var(lhs, zero)?;
+            // The value is represented internally as one with its sign stored separately
+            // As we are within signed division we can just check whether we have one here.
+            let rhs_is_one = self.eq_var(rhs, one)?;
+            let both_true = self.and_var(lhs_is_zero, rhs_is_one, AcirType::unsigned(1))?;
```

Comment:
Shouldn't this be `AcirType::signed`?

Reply:
I am comparing `lhs_is_zero` and `rhs_is_one` which should be two booleans. We could also just do a multiplication here, and in fact that is what `and_var` will do internally for bit size one, I just thought this read more clearly with `and`.

Reply:
Ah, of course! 😵‍💫